src/lava/lib/optimization/apps/scheduler/solver.py

```diff
@@ -0,0 +1,289 @@
+    def lava_backend(self):
+        return self._lava_backend
+
+    @lava_backend.setter
+    def lava_backend(self, backend: str):
+        self._lava_backend = backend
+
+    @property
+    def lava_solver_report(self):
+        return self._lava_solver_report
+
+    @property
+    def lava_solution(self):
+        return self._lava_solution
+
+    def gen_qubo_mat(self):
+        adj_mat = self.problem.adjacency
+        self._qubo_matrix = MISProblem._get_qubo_cost_from_adjacency(
+            adj_mat, self.qubo_weights[0], self.qubo_weights[1])
+
+    def gen_qubo_problem(self):
+        self.gen_qubo_mat()
+        self._qubo_problem = QUBO(self.qubo_matrix)
+
+    def solve_with_netx(self):
+        """ Find an approximate maximum independent set using networkx. """
+        solution = maximum_independent_set(self.graph)
+        solution = np.array(list(solution))
+        self._netx_solution = np.zeros((solution.size, 4))
+        nds = self.graph.nodes
+        for j, sol_node in enumerate(solution):
+            agent_id = nds[sol_node]["agent_id"]
+            task_attrs = nds[sol_node]["task_attr"]
+            self._netx_solution[j, :] = (
+                np.hstack((sol_node, agent_id, task_attrs)))
+
+    def solve_with_lava_qubo(self, timeout=1000):
+        """ Find a maximum independent set using QUBO in Lava. """
+        self.gen_qubo_problem()
+        solver = OptimizationSolver(self.qubo_problem)
+        self._lava_solver_report = solver.solve(
+            config=SolverConfig(
+                timeout=timeout,
+                hyperparameters=self.qubo_hyperparams,
+                target_cost=self.qubo_target_cost,
+                backend=self.lava_backend,
+                probe_cost=self.probe_cost,
+                probe_time=self.probe_loihi_exec_time,
+                probe_energy=self.probe_loihi_energy,
+                log_level=40
+            )
+        )
+        qubo_state = self.lava_solver_report.best_state
+        solution = (
+            np.array(self.graph.nodes))[np.where(qubo_state)[0]]
+        self._lava_solution = np.zeros((solution.size, 4))
+        nds = self.graph.nodes
+        for j, sol_node in enumerate(solution):
+            agent_id = nds[sol_node]["agent_id"]
+            task_attrs = nds[sol_node]["task_attr"]
+            self._lava_solution[j, :] = (
+                np.hstack((sol_node, agent_id, task_attrs)))
+
+
+class SatelliteScheduler(Scheduler):
+    def __init__(self,
+                 ssp: SatelliteScheduleProblem,
+                 **kwargs):
+        qubo_weights = kwargs.pop("qubo_weights", (1, 8))
+        probe_cost = kwargs.pop("probe_cost", False)
+        super(SatelliteScheduler, self).__init__(ssp,
+                                                 qubo_weights=qubo_weights,
+                                                 probe_cost=probe_cost,
+                                                 **kwargs)
+        self.num_satellites = ssp.num_satellites
```

Comment:
The definition of self.num_satellites would be a great application for a setter/getter.

Reply:
It's an alias. Users should not set it from outside. I could add a getter, but that's too much effort :)